File: OOP/03_05/autodata/cpp/object/cat.cc
#include <iostream>
#include "cat.h"

namespace Edu {
    namespace Brooklyn {

        using std::string;
        using std::cout;
        using std::endl;

        Cat::Cat() {
            this->name = "cat who does not yet to have a name";
        }

        Cat::Cat(string name) {
            this->name = name;
        }

        void Cat::tap(const Cat& other) const {
            cout << this->name << " is tapping " << other.name << endl;
        }

        string Cat::getName() const {
            return name;
        }
        
    }
}

File: OOP/03_05/dynamicdata/java/HeapCatApp/src/main/java/edu/brooklyn/Cat.java
package edu.brooklyn;

public class Cat {
	private String name;

	public Cat() {
		this.name = "cat who does not yet to have a name";
	}

	public Cat(String name) {
		this.name = name;
	}

	public void tap(Cat other) {
		System.out.println(this.name + " is tapping " + other.name);
	}

	public String getName() {
		return name;
	}
}
